docs/howto/use_modules_and_layers.md

```diff
@@ -0,0 +1,392 @@
+```
+
+We encountered two new data types `Has` and `ZLayer`, let's get familiar with them.
+
+### The `Has` data type
+
+`Has[A]` represents a dependency on a service of type `A`. Two `Has[_]` can be combined _horizontally_ through `+` and `++` operators, as in
+
+```scala mdoc:silent
+val repo: Has[Repo.Service] = ???
+val logger: Has[Logger.Service] = ???
+
+val mix: Has[Repo.Service] with Has[Logger.Service] = repo ++ logger
+``` 
+
+At this point you might ask: what's the use of `Has` if the resulting type is still a mix of two traits? The extra power given by `Has` is that the resulting data structure is backed by an _etherogeneus map_ from service type to service implementation, that collects each instance that is mixed in so that the instances can be accessed/extracted/modified individually, still guaranteeing supreme type safety.
+
+```scala mdoc:silent
+// get back the repo from the mixed value:
+val log = mix.get[Has[Logger.Service]].log("Hello modules!")
+```
+
+As per the recipe above, it is extremely convenient to define a type alias for `Has[Service]`.
+Usually we don't create a `Has` directly, but we do that through `ZLayer`.
+
+### The `ZLayer` data type
+
+`ZLayer[-RIn, +E, +ROut <: Has[_]]` is a recipe to build an environment of type `ROut`, starting from a value `RIn`, possibly producing an error `E` during creation. 
+
+In adherence with environmental concepts, the absence of a required input is represented by `RIn = Any`, conveniently  used in the type alias `ZLayer#NoDeps`.
+
+The `ZLayer` companion object offers a number of constructors to build layers (from pure values, from managed resources to guarantee acquire/release, from other services). 
+
+We can compose `layerA` and `layerB`  _horizontally_ to build a layer that has the requirements of both layers, to provide the capabilities of both layers, through `layerA ++ layerB`
+ 
+We can also compose layers _vertically_, meaning the output of one layer is used as input for the subsequent layer to build the next layer, resulting in one layer with the requirement of the first and the output of the second layer: `layerA >>> layerB` 
+
+### Example: wiring modules together
+Here we define a module to cope with CRUD operations for the `User` domain object. We provide also a live implemntation of the module that depends on a sql connection.
+
+```scala mdoc:silent
+object  userRepo {
+  trait Service {
+    def getUser(userId: UserId): IO[DBError, Option[User]]
+    def createUser(user: User): IO[DBError, Unit]
+  }
+
+  type UserRepo = Has[Service]
+
+  // This simple live version depends only on a DB Connection
+  val inMemory: ZLayer.NoDeps[Nothing, UserRepo] = ZLayer.succeed(
+      new Service {
+        def getUser(userId: UserId): IO[DBError, Option[User]] = ???
+        def createUser(user: User): IO[DBError, Unit] = ???
+      }
+    )
+
+  //accessor methods
+  def getUser(userId: UserId): ZIO[UserRepo, DBError, Option[User]] =
+    ZIO.accessM(_.get.getUser(userId))
```

Comment:
I think you need a type annotation on `get` here.

Reply:
No, if the environment is a single `Has[A]` you don't need type hinting the compiler. It will infer correctly that `_.get` returns type `A`. In case of composite environment `Has[A] with Has[B]` you must specify what you want.